modules/php/States/PlayCardTrait.php

```diff
@@ -40,6 +42,25 @@ public function st_playCard()
     $playRule = $game->getGameStateValue("playRule");
     $cardsPlayed = $game->getGameStateValue("playedCards");
 
+    // check bonus rules
+    $addInflation = Utils::getActiveInflation() ? 1 : 0;
+    $partyBonus =
+      Utils::getActivePartyBonus() && Utils::isPartyInPlay()
+        ? 1 + $addInflation
+        : 0;
+    if ($partyBonus > 0) {
+      RulePartyBonus::notifyActiveFor($player_id);
+    }
+    $richBonus =
+      Utils::getActiveRichBonus() && Utils::hasMostKeepers($player_id)
+        ? 1 + $addInflation
+        : 0;
+    if ($richBonus > 0) {
+      RuleRichBonus::notifyActiveFor($player_id);
+    }
+    
+    $playRule += $addInflation + $partyBonus + $richBonus;
```

Comment:
I believe there is an issue here with "play all but 1". We need to make sure $playRule != 1 before adding the inflation, partyBonus & richBonus.

Reply:
Actually, the "play-all-but-1" has another rule that we don't enforce at the moment:
"If you started with no cards and only drew 1".
I think I'll fix this issue in a different branch for you to review.

Reply:
Also even more complicated actually: "Play All but 1" has a number on it, so with Inflation applied it will actually become "Play All but 2"!

Reply:
See my comment above, there is indeed more complications to this.
Maybe I would suggest to merge & include this branch as it is, to have as much as possible in the alfa release.
And consider this as additional changes in a separate branch for Rich & Poor.

Reply:
Yes! And Play All but 2 + rich bonus = play all...
I'll add a comment in the code.